package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1039,11 +1039,17 @@ def principal_axes(group, pbc=False):
         """
         atomgroup = group.atoms
         e_val, e_vec = np.linalg.eig(atomgroup.moment_of_inertia(pbc=pbc))
-
+        
         # Sort
         indices = np.argsort(e_val)[::-1]
-        # Return transposed in more logical form. See Issue 33.
-        return e_vec[:, indices].T
+        # Make transposed in more logical form. See Issue 33.
+        e_vec = e_vec[:, indices].T
+
+        # Make sure the right hand convention is followed
+        if np.dot(np.cross(e_vec[0], e_vec[1]), e_vec[2]) < 0:
+            e_vec *= -1
```

Comment:
Multiply all of them or only `evec[2]`  – either one works, not sure if one is preferable over the other.